tests/conftest.py

```diff
@@ -229,6 +229,12 @@ def empty_config(tmp_path) -> Config:
         configuration.CONFIG = current_config
         manager.load_profile(current_profile_name)
 
+        # This will set important global constants that are based on the location of the config folder. Without this,
+        # things like the :class:`aiida.engine.daemon.client.DaemonClient` will not function properly because the paths
+        # of the daemon files will be incorrect and still pointing to the path of the temporary config folder created by
+        # this fixture.
+        settings.set_configuration_directory()
```

Comment:
Looking into this function, I have a few small suggestions:

 * could you please update its docstring? The name and docstring suggests that this function just sets the docstring, but in reality it also creates the directories.
 * could you please add a `directory` argument that defaults to `None`, and use this function furhter up instead of the lines below? This would make it clearer that we are undoing here what was done before
```
    # Create a temporary folder, set it as the current config directory path and reset the loaded configuration
    settings.AIIDA_CONFIG_FOLDER = str(tmp_path)
    # Create the instance base directory structure, the config file and a dummy profile
    settings.create_instance_directories()
```
 * maybe move the call here before the `load_profile` (I guess it doesn't make a difference but seems more logical).